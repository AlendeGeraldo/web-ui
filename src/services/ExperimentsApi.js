// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';
// projects path
const projectsPath = '/projects';
// api object
const experimentsApi = axios.create({
  baseURL: `${URL}${projectsPath}`,
});
// experiments path
const experimentsPath = '/experiments';

// API METHODS
/**
 * List Experiments
 * @param {string} projectId
 * @returns {Promise}
 */
const listExperiments = (projectId) =>
  new Promise((resolve, reject) => {
    // requesting projects
    experimentsApi
      .get(`/${projectId}${experimentsPath}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

// EXPORT DEFAULT
export default {
  listExperiments,
};

// export const getExperimentList = async (id) => {
//   try {
//     const response = await projectsApi.get(`/projects/${id}/experiments`);
//     return response;
//   } catch (error) {
//     message.error(error.message);
//   }
// };

// export const getExperiment = async (projectId, experimentId) => {
//   try {
//     const response = await projectsApi.get(
//       `/projects/${projectId}/experiments/${experimentId}`
//     );
//     return response;
//   } catch (error) {
//     message.error(error.message);
//   }
// };

// export const createExperiment = async (id, experimentName) => {
//   try {
//     const body = {
//       name: experimentName,
//     };
//     const response = await projectsApi.post(
//       `/projects/${id}/experiments/`,
//       body
//     );
//     return response;
//   } catch (error) {
//     message.error(error.message);
//   }
// };

// export const updateExperiment = async (projectId, experimentId, body) => {
//   try {
//     const response = await projectsApi.patch(
//       `/projects/${projectId}/experiments/${experimentId}`,
//       body
//     );
//     return response;
//   } catch (error) {
//     if (error.response) message.error(error.response.data.message);
//   }
// };

// // \"name\": \"Auto-featuring Experiment\",
// // \"pipelineIdTrain\": \"23266cfd-4ed6-43d6-b8a0-ca8440d251c6\",
// // \"pipelineIdDeploy\": \"fe5205f5-7f76-4f57-84ca-ea6dd62670e8\",
// // \"datasetId\": \"0a10c0ac-ff3b-42df-ab7a-dc2962a1750c\",
// // \"headerId\": \"482b603f-23c1-4a10-9b79-8c5b91c6c0cb\",
// // \"targetColumnId\": \"3191a035-97a6-4e29-90d4-034cb1f87237\",
// // \"parameters\": \"{ price: 8, auto-featuring: true }\",
// // \"position\": 1
