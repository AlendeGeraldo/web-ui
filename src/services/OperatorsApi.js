// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';
// projects path
const projectsPath = '/projects';
// api object
const operatorsApi = axios.create({
  baseURL: `${URL}${projectsPath}`,
});
// experiments path
const experimentsPath = '/experiments';
// operators path
const operatorsPath = '/operators';

// API METHODS
/**
 * List Operators
 * @param {string} projectId
 * @returns {Promise}
 */
const listOperators = (projectId, experimentId) =>
  new Promise((resolve, reject) => {
    // requesting operators
    operatorsApi
      .get(`/${projectId}${experimentsPath}/${experimentId}${operatorsPath}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

/**
 * Detail Operator
 * @param {string} projectId
 * @returns {Promise}
 */
/* const detailOperator = (projectId, operatorId) =>
  new Promise((resolve, reject) => {
    // requesting operator
    operatorsApi
      .get(`/${projectId}${operatorsPath}/${operatorId}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

/**
 * Create Operator
 * @param {string} projectId
 * @param {string} operatorName
 * @returns {Promise}
 */
/* const createOperator = (projectId, operatorName) =>
  new Promise((resolve, reject) => {
    // creating body object
    const body = {
      name: operatorName,
    };

    // creating operator
    operatorsApi
      .post(`/${projectId}${operatorsPath}`, body)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

/**
 * Update Operator
 * @param {string} projectId
 * @param {string} operatorId
 * @param {Object} operator
 * @returns {Promise}
 */
/* const updateOperator = (projectId, operatorId, operator) =>
  new Promise((resolve, reject) => {
    // updating operator
    operatorsApi
      .patch(`/${projectId}${operatorsPath}/${operatorId}`, operator)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

/**
 * Delete Operator
 * @param {string} projectId
 * @param {string} operatorId
 * @returns {Promise}
 */
/* const deleteOperator = (projectId, operatorId) =>
  new Promise((resolve, reject) => {
    // deleting operator
    operatorsApi
      .delete(`/${projectId}${operatorsPath}/${operatorId}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

// EXPORT DEFAULT
export default {
  listOperators,
  /*   detailOperator,
  createOperator,
  updateOperator,
  deleteOperator, */
};
