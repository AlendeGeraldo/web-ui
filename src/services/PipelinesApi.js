// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PIPELINES_API || 'http://localhost:3000';
// api object
const pipelinesApi = axios.create({
  baseURL: `${URL}`,
});
// train path
const trainPath = '/train';

/**
 * Train Experiment
 * @param {Object} trainObject
 * @returns {Promise}
 */
const trainExperiment = (trainObject) =>
  new Promise((resolve, reject) => {
    // training experiment
    pipelinesApi
      .post(`${trainPath}`, trainObject)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

// EXPORT DEFAULT
export default {
  trainExperiment,
};
