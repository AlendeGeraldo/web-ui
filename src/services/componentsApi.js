/* eslint-disable consistent-return */
import axios from 'axios';
import { message } from 'antd';

export const URL =
  process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';

export const uploadUrl = `${URL}/components/upload`;

export const downloadUrl = `${URL}/components/download`;

export const componentsApi = axios.create({
  baseURL: URL,
});

export const getAllComponents = async () => {
  try {
    const response = await componentsApi.get(`/components`);
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const getComponent = async (id) => {
  try {
    const response = await componentsApi.get(`/components/${id}`);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const createComponent = async (componentName) => {
  try {
    const body = {
      name: componentName,
    };
    const response = await componentsApi.post(`/components/`, body);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const updateComponent = async (id, componentName) => {
  try {
    const body = {
      name: componentName,
    };
    const response = await componentsApi.patch(`/components/${id}`, body);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const updateParameters = async (id, componentParameters) => {
  try {
    const body = {
      parameters: componentParameters,
    };
    const response = await componentsApi.patch(`/components/${id}`, body);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const deleteFiles = async (id, componentFiles) => {
  try {
    const body = {
      files: componentFiles,
    };
    const response = await componentsApi.post(
      `/components/deleteFiles/${id}`,
      body
    );
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const deleteComponent = async (id) => {
  try {
    const response = await componentsApi.delete(`/components/${id}`);
    return response;
  } catch (error) {
    message.error(error.message);
  }
};
