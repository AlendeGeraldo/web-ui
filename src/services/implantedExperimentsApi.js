// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PIPELINES_API || 'http://localhost:3000';
// api object
const pipelinesApi = axios.create({
  baseURL: `${URL}`,
});
const deploymentsPath = '/deployments';

/**
 * Get Deployed Experiments
 * @returns {Promise}
 */
const getDeployedExperiments = () =>
  new Promise((resolve, reject) => {
    // get deployed experiment
    pipelinesApi
      .get(deploymentsPath)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

// EXPORT DEFAULT
export default {
  getDeployedExperiments,
};
