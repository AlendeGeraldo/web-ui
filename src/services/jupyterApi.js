/* eslint-disable consistent-return */
import { message } from 'antd';
import axios from 'axios';
import uuidv4 from 'uuid/v4';
import Cookies from 'universal-cookie';

import * as componentsServices from './componentsApi';

export const URL = process.env.REACT_APP_JUPYTER_API || '';

export const componentsApi = axios.create({
  baseURL: URL,
});

export const getNamespaces = async () => {
  try {
    const response = await componentsApi.get(`/jupyter/api/namespaces`);
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const getNotebook = async (namespace) => {
  try {
    const response = await componentsApi.get(
      `/jupyter/api/namespaces/${namespace}/notebooks`
    );
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const uploadFile = async (id, namespace, notebook, fileName) => {
  try {
    const responseFileBase64 = await componentsServices.downloadBase64(
      id,
      fileName
    );

    if (!responseFileBase64) {
      return;
    }

    const { payload } = responseFileBase64.data;
    const body = {
      content: payload,
      format: 'base64',
      name: fileName,
      path: fileName,
      type: 'file',
    };

    const uuid = uuidv4();
    const cookies = new Cookies();
    cookies.set('_xsrf', uuid, { path: '/' });

    const response = await componentsApi.put(
      `/notebook/${namespace}/${notebook}/api/contents/${fileName}`,
      body,
      {
        credentials: 'include',
        headers: {
          'X-XSRFToken': uuid,
        },
      }
    );
    return response;
  } catch (error) {
    message.error(error.message);
  }
};
