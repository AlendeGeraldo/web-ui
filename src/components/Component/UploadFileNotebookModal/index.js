/* eslint-disable react/forbid-prop-types */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

import PropTypes from 'prop-types';

import { Form, Modal, message, Select, Spin } from 'antd';

import * as jupyterServices from '../../../services/jupyterApi';

const { Option } = Select;

/* 
  This function is used to check if form has errors
*/
const hasErrors = (fieldsError) => {
  return Object.keys(fieldsError).some((field) => fieldsError[field]);
};

class UploadFileNotebookModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      listNotebook: [],
    };
  }

  sleep = (milliseconds) => {
    return new Promise((resolve) => setTimeout(resolve, milliseconds));
  };

  onCancelModal = () => {
    const { onCancel } = this.props;
    this.setState({ listNotebook: [] });
    onCancel();
  };

  onChangeNamespace = async (namespaceSelected) => {
    this.setState({ loading: true });
    const response = await jupyterServices.getNotebook(namespaceSelected);

    let optionsNotebooks;
    if (response) {
      const { notebooks } = response.data;
      if (notebooks && notebooks.length > 0) {
        optionsNotebooks = notebooks.map((notebook) => (
          <Option key={notebook.name} value={notebook.name}>
            {notebook.name}
          </Option>
        ));
      } else {
        optionsNotebooks = [
          <Option
            key='new-notebook'
            value='new-notebook'
            onClick={() => {
              window.open('/jupyter/');
            }}
          >
            <a>Criar notebook</a>
          </Option>,
        ];
      }
    }
    this.setState({
      listNotebook: optionsNotebooks,
    });
    this.props.form.setFieldsValue({
      notebook: '',
    });
    this.setState({ loading: false });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.form.validateFields(async (err, values) => {
      if (err) {
        return;
      }

      const { filePath, fileName } = this.props;
      const { namespace, notebook } = values;
      this.setState({ loading: true });

      const response = await jupyterServices.uploadFile(
        namespace,
        notebook,
        filePath,
        fileName
      );

      if (response) {
        message.success(`Upload realizado com sucesso.`);
        this.onCancelModal();
        await this.sleep(1000);
        window.open(`/notebook/${namespace}/${notebook}/notebooks/${fileName}`);
      }

      this.setState({ loading: false });
    });
  };

  render() {
    const { namespaces, visible, form } = this.props;
    const { getFieldDecorator, getFieldsError } = form;
    const { listNotebook } = this.state;

    let optionsNamespaces;
    if (namespaces) {
      optionsNamespaces = namespaces.map((namespace) => (
        <Option key={namespace.namespace} value={namespace.namespace}>
          {namespace.namespace}
        </Option>
      ));
    }

    const formUI = (
      <Form layout='vertical'>
        <Form.Item label='Namespaces: '>
          {getFieldDecorator('namespace', {
            rules: [{ required: true, message: ' ' }],
          })(
            <Select onChange={this.onChangeNamespace}>
              {optionsNamespaces}
            </Select>
          )}
        </Form.Item>
        <Form.Item label='Notebooks: '>
          {getFieldDecorator('notebook', {
            rules: [{ required: true, message: ' ' }],
          })(<Select>{listNotebook}</Select>)}
        </Form.Item>
      </Form>
    );

    return (
      <Modal
        visible={visible}
        title='Editar arquivo no jupyter notebook'
        okText='Editar'
        cancelText='Cancelar'
        onCancel={this.onCancelModal}
        onOk={this.handleSubmit}
        okButtonProps={{ disabled: hasErrors(getFieldsError()) }}
      >
        <Spin spinning={this.state.loading} delay={100}>
          {formUI}
        </Spin>
      </Modal>
    );
  }
}

UploadFileNotebookModal.propTypes = {
  filePath: PropTypes.string.isRequired,
  fileName: PropTypes.string.isRequired,
  namespaces: PropTypes.arrayOf(PropTypes.object).isRequired,
  visible: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  form: PropTypes.shape({
    getFieldDecorator: PropTypes.func,
    getFieldsError: PropTypes.func,
  }).isRequired,
};

export default Form.create({ name: 'upload-file-notebook' })(
  UploadFileNotebookModal
);
