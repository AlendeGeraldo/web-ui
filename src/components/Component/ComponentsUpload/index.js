/* eslint-disable react/forbid-prop-types */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Icon, message, Upload } from 'antd';
import UploadFileNotebookModal from '../UploadFileNotebookModal';
import * as componentsServices from '../../../services/componentsApi';

const { Dragger } = Upload;

const ComponentsUpload = (props) => {
  const { details, namespaces, updateComponentFile } = props;
  const [modalIsVisible, setModalIsVisible] = useState(false);

  const showModal = () => {
    setModalIsVisible(true);
  };

  const hideModal = () => {
    // const { form } = this.formRef.props;

    // this.setState({ modalIsVisible: false });

    // form.resetFields();
    setModalIsVisible(false);
  };

  const DraggerProps = {
    multiple: false,
    action: `${componentsServices.uploadUrl}/${details.uuid}`,
    defaultFileList: details.file ? [details.file] : [],
    beforeUpload: (file) => {
      const auxFile = file;
      auxFile.path = `components/${details.uuid}/${file.name}`;
      auxFile.url = `${componentsServices.downloadUrl}/${details.uuid}/${file.name}`;
    },
    onChange(info) {
      const { status } = info.file;
      if (status === 'done') {
        message.success(`${info.file.name} salvo com sucesso.`);
        updateComponentFile(info.file);
      } else if (status === 'removed') {
        if (!info.file.error) {
          message.success(`${info.file.name} removido com sucesso.`);
        }
      } else if (status === 'error') {
        if (info.file.error.status === 400) {
          message.error(`Só é possível realizar o upload de um arquivo!`);
        } else {
          message.error(`Falha no upload do arquivo ${info.file.name}`);
        }
      }
    },
    onRemove: async (file) => {
      if (!file.error) {
        const response = await componentsServices.deleteFiles(details.uuid, [
          file.name,
        ]);
        if (!response) {
          return false;
        }
      }
      updateComponentFile(null);
      return true;
    },
  };

  return (
    <>
      <UploadFileNotebookModal
        filePath={details.file ? details.file.path : ''}
        fileName={details.file ? details.file.name : ''}
        namespaces={namespaces}
        visible={modalIsVisible}
        onCancel={hideModal}
      />
      <Dragger {...DraggerProps}>
        <p className='ant-upload-drag-icon'>
          <Icon type='inbox' />
        </p>
        <p className='ant-upload-text'>
          Clique ou arraste o arquivo para esta área para fazer o upload
        </p>
      </Dragger>
      <br />
      <Button
        className='float-right'
        style={
          details.file
            ? { float: 'right' }
            : { display: 'none', float: 'right' }
        }
        onClick={showModal}
        type='primary'
      >
        Editar arquivo
      </Button>
    </>
  );
};

ComponentsUpload.propTypes = {
  details: PropTypes.shape({
    uuid: PropTypes.string,
    file: PropTypes.shape({
      name: PropTypes.string,
      path: PropTypes.string,
      status: PropTypes.string,
      uid: PropTypes.string,
      url: PropTypes.string,
    }),
  }).isRequired,
  namespaces: PropTypes.arrayOf(
    PropTypes.shape({
      namespace: PropTypes.string,
    })
  ).isRequired,
  updateComponentFile: PropTypes.func.isRequired,
};

export default ComponentsUpload;
