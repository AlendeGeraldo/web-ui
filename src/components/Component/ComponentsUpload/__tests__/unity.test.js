import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import nock from 'nock';
import { Button, message, Upload } from 'antd';
import { URL } from '../../../../services/componentsApi';
import UploadFileNotebookModal from '../../UploadFileNotebookModal';
import ComponentsUpload from '..';

const { Dragger } = Upload;

const details = {
  uuid: '1',
  createdAt: '2019-11-05T17:16:53.000Z',
  name: 'AutoML',
  parameters: [
    { name: 'b', type: 'int', required: true },
    { name: 'a', type: 'int', required: false },
  ],
};

const namespaces = [
  {
    namespace: 'anonymous',
    role: 'contributor',
    user: 'anonymous@kubeflow.org',
  },
];

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  component: {
    details,
    namespaces,
  },
});

describe('ComponentsUpload component', () => {
  it('is expected render without crashing', () => {
    shallow(
      <Provider store={store}>
        <ComponentsUpload />
      </Provider>
    );
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <ComponentsUpload />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('toggleModal', () => {
    const wrapper = mount(
      <Provider store={store}>
        <ComponentsUpload />
      </Provider>
    );
    const button = wrapper.find(Button);
    button.simulate('click');
    const modal = wrapper.find(UploadFileNotebookModal);
    expect(modal.props().visible).toBe(true);
  });

  it('DraggerProps', async () => {
    const wrapper = mount(
      <Provider store={store}>
        <ComponentsUpload />
      </Provider>
    );

    const dragger = wrapper.find(Dragger);
    const draggerProps = dragger.props();

    const spyMsgSuccess = jest
      .spyOn(message, 'success')
      .mockImplementation(() => {});
    const spyMsgError = jest
      .spyOn(message, 'error')
      .mockImplementation(() => {});

    // call beforeUpload without crashing
    draggerProps.beforeUpload({ name: 'name' });

    // check onChange with status done
    draggerProps.onChange({ file: { name: 'name', status: 'done' } });
    expect(spyMsgSuccess).toHaveBeenCalledWith(`name salvo com sucesso.`);

    // check onChange with status removed and error
    draggerProps.onChange({
      file: { status: 'removed', error: { status: 400 } },
    });

    // check onChange with status removed without error
    draggerProps.onChange({ file: { name: 'name', status: 'removed' } });
    expect(spyMsgSuccess).toHaveBeenCalledWith(`name removido com sucesso.`);

    // check onChange with status error with response 400
    draggerProps.onChange({
      file: { name: 'name', status: 'error', error: { status: 400 } },
    });
    expect(spyMsgError).toHaveBeenCalledWith(
      `Só é possível realizar o upload de um arquivo!`
    );

    // check onChange with status error with response 500
    draggerProps.onChange({
      file: { name: 'name', status: 'error', error: { status: 500 } },
    });
    expect(spyMsgError).toHaveBeenCalledWith(`Falha no upload do arquivo name`);

    // check onRemove success
    nock(URL)
      .post(`/components/deleteFiles/${details.uuid}`)
      .reply(200);
    await draggerProps
      .onRemove({
        file: { name: 'name' },
      })
      .then((response) => {
        expect(response).toBe(true);
      });

    // check onRemove error
    nock(URL)
      .post(`/components/deleteFiles/${details.uuid}`)
      .reply(400);
    await draggerProps
      .onRemove({
        file: { name: 'name' },
      })
      .then((response) => {
        expect(response).toBe(false);
      });

    // check onRemove with file error
    draggerProps.onRemove({
      file: { error: { status: 500 } },
    });
  });
});
