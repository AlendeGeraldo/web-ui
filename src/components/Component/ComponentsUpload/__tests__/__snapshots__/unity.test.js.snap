// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentsUpload component is expected render html correctly 1`] = `
<Fragment>
  <Form(UploadFileNotebookModal)
    fileName=""
    filePath=""
    namespaces={
      Array [
        Object {
          "namespace": "anonymous",
          "role": "contributor",
          "user": "anonymous@kubeflow.org",
        },
      ]
    }
    onCancel={[Function]}
    visible={false}
  />
  <Dragger
    action="http://localhost:3000/components/upload/1"
    beforeUpload={[Function]}
    defaultFileList={Array []}
    multiple={false}
    onChange={[Function]}
    onRemove={[Function]}
  >
    <p
      className="ant-upload-drag-icon"
    >
      <Icon
        type="inbox"
      />
    </p>
    <p
      className="ant-upload-text"
    >
      Clique ou arraste o arquivo para esta Ã¡rea para fazer o upload
    </p>
  </Dragger>
  <br />
  <Button
    block={false}
    className="float-right"
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
        "float": "right",
      }
    }
    type="primary"
  >
    Editar arquivo
  </Button>
</Fragment>
`;
